projects/admin/src/app/deposit/upload/upload.component.ts

```diff
@@ -370,14 +360,37 @@ export class UploadComponent implements OnInit, AfterContentChecked {
    * @param file File data
    */
   private addFormField(file: any) {
-    this.filesForm.push(
```

Comment:
This can be perhaps done with a custom editor component inspired from:
https://github.com/ngx-formly/ngx-formly/blob/92bb6d70388f76725f6f7fcdde4febee42212dd1/src/core/src/lib/templates/field-array.type.ts#L56

But I'm not sure to understand well.